### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by ericgaspar**
## Problem

- *Description of why you made this PR*

## Solution

- *And how do you fix that problem*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

**Comment by Ddataa** [comment]
@ericgaspar 
I am trying to test this for real now

Can you please tell me what there is new to test here ?

apart from being the latest cryptpad version...

**Comment by Ddataa** [comment]
installing from scratch (in a clean yunohost) is working but the checkup page indicate some new errors regarding HTTP headers...
tests 17/18/21/24/40

I still have to test upgrade from the previous version

**Comment by ericgaspar** [comment]
!testme

**Comment by Ddataa** [comment]
We have to solve those checkup issues before we can merge otherwise it is just calling for later trouble

**Comment by Ddataa** [comment]
also it would be nice to be able to keep the config when upgrade

**Comment by ericgaspar** [comment]
!testme

File: conf/nginx.conf [hunk 1]
```diff
@@ -4,88 +4,179 @@ set $allowed_origins "https://${sandbox_domain}";
 set $api_domain "__DOMAIN__";
 set $files_domain "__DOMAIN__";
 ssl_ecdh_curve secp384r1;
-more_set_headers "Access-Control-Allow-Origin: '${allowed_origins}'";
+
+more_set_headers "X-XSS-Protection: 1; mode=block";
+more_set_headers "X-Content-Type-Options: nosniff";
```
Comment (alexAubin) [inline conf/nginx.conf:9]:
```suggestion
```

These are already part of Yunohost's conf : https://github.com/YunoHost/yunohost/blob/dev/conf/nginx/security.conf.inc#L33

**Comment by alexAubin** [comment]
>the checkup page indicate some new errors regarding HTTP headers...

Could you elaborate what are the errors ? x_x

**Comment by Tagadda** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by Ddataa** [comment]
out-of-the-box nothing works...
:-1: 

installed on a fresh VPS with fresh yunohost install
Checkup has 43 / 55 tests passed.

Tests Failed at :

Test 06:
```
Your browser was not able to load an iframe using the origin specified as httpSafeOrigin (https://sandbox.mydomain.xyz) in cryptpad/config/config.js. This can be caused by an invalid httpUnsafeDomain, invalid CSP configuration in your reverse proxy, invalid SSL certificates, and many other factors. More information about your particular error may be found in your browser console. Changes to cryptpad/config/config.js will require a server restart in order for [/api/config](https://mydomain.xyz/api/config) to be updated.
```

Test 17:
```
/sheet/inner.html does not have the required 'content-security-policy' headers set. This is most often related to incorrectly configured sandbox domains or reverse proxies.
```

Test 18:
```
/common/onlyoffice/v5/web-apps/apps/spreadsheeteditor/main/index.html does not have the required 'content-security-policy' headers set. This is most often related to incorrectly configured sandbox domains or reverse proxies.
```

Test 21:
```
/api/config returned an HTTP status code other than 200 when accessed from the sandbox domain.
```

Test 24:
```
https://mydomain.xyz/sheet/inner.html was served with incorrect Content-Security-Policy headers.

A value of "'none'" was expected for the default-src directive.

A value of "'unsafe-inline' 'self' https://mydomain.xyz" was expected for the style-src directive.

A value of "'self' data: https://mydomain.xyz" was expected for the font-src directive.

A value of "https://mydomain.xyz" was expected for the child-src directive.

A value of "'self' blob: https://sandbox.mydomain.xyz" was expected for the frame-src directive.

A value of "'self' resource: https://mydomain.xyz 'unsafe-eval' 'unsafe-inline'" was expected for the script-src directive.

A value of "'self' blob: https://mydomain.xyz https://sandbox.mydomain.xyz wss://mydomain.xyz" was expected for the connect-src directive. This rule restricts which URLs can be loaded by scripts. Overly permissive settings can allow users to be tracked using external resources, while overly restrictive settings may block pages from loading entirely.

A value of "'self' data: blob: https://mydomain.xyz" was expected for the img-src directive.

A value of "blob:" was expected for the media-src directive.

A value of "'self' https://mydomain.xyz" was expected for the frame-ancestors directive. This rule determines which sites can embed content from this instance in an iframe.

A value of "'self'" was expected for the worker-src directive.
```

Test 25:
```
https://mydomain.xyz/ was served with incorrect Content-Security-Policy headers.

A value of "'none'" was expected for the default-src directive.

A value of "'unsafe-inline' 'self' https://mydomain.xyz" was expected for the style-src directive.

A value of "'self' data: https://mydomain.xyz" was expected for the font-src directive.

A value of "https://mydomain.xyz" was expected for the child-src directive.

A value of "'self' blob: https://sandbox.mydomain.xyz" was expected for the frame-src directive.

A value of "'self' resource: https://mydomain.xyz" was expected for the script-src directive.

A value of "'self' blob: https://mydomain.xyz https://sandbox.mydomain.xyz wss://mydomain.xyz" was expected for the connect-src directive. This rule restricts which URLs can be loaded by scripts. Overly permissive settings can allow users to be tracked using external resources, while overly restrictive settings may block pages from loading entirely.

A value of "'self' data: blob: https://mydomain.xyz" was expected for the img-src directive.

A value of "blob:" was expected for the media-src directive.

A value of "'self' https://mydomain.xyz" was expected for the frame-ancestors directive. This rule determines which sites can embed content from this instance in an iframe.

A value of "'self'" was expected for the worker-src directive.
```

Test 40:
```
A placeholder file was expected to be available at https://mydomain.xyz/block/placeholder.txt, but it was not found. This commonly indicates a mismatch between the API server's blockPath value and the path that the webserver or reverse proxy is attempting to serve. This misconfiguration will cause errors with login, registration, and password change.
```

Plus minor warning on some other tests...

The App is dysfunctionnal after the installation is done, the frontpage open but no app can be openned, the webbrowser gives this error message when trying to open an app :

```
LibreWolf Can’t Open This Page

To protect your security, sandbox.mydomain.xyz will not allow LibreWolf to display the page if another site has embedded it. To see this page, you need to open it in a new window.
 ```
Same behavior with Firefox and Chromium...

**Comment by Ddataa** [comment]
I dont know from what end to take those 45 commits to be tested....

To be honest we shall split them into small PR for each so we can iterate and test each
@mathilde-cryptpad  what do you think and can you help me split and create PRs, I can do testing of them...

**Pull request closed without merging**